src/test/java/tahafurkan/sandbox/usermanagement/service/UserServiceTest.java

```diff
@@ -92,4 +102,126 @@ void create_GivenAllowedUsername_ReturnsSavedUser() {
+            int id = 1;
+            User user = new User();
+            user.setId(id);
+            userService.get(id);
+        });
+    }
+
+    @Test
+    void get_WithAnIdExists_ReturnUser() {
+        int id = 1;
+        User user = new User();
+        user.setId(id);
+        given(userRepository.findById(id)).willReturn(Optional.of(user));
+        UserDto userDto = userService.get(id);
+        assertNotNull(userDto);
+        assertEquals(1, userDto.getId());
+    }
+
+    @Test
+    void getAll_ReturnsEmptyUserList() {
+        List<UserDto> userList = userService.getAll();
+        Mockito.verify(userRepository, times(1)).findAll();
+        assertEquals(0, userList.size());
+
+
+    }
+
+    @Test
+    void getAll_ReturnsPresentUserList() {
+        List<User> records = new ArrayList<>();
+
+        List<Address> addressList = new ArrayList<>();
+        addressList.add(new Address(1, "home", "Istanbul", "Turkey"));
+        User user_1 = new User(1, "Taha", "Unsal", "taha.f.unsal@gmail.com", "taha.furkan", addressList);
+        User user_2 = new User(2, "Zehra", "Unsal", "zehra.unsal@gmail.com", "zehra.unsal", addressList);
+        records.add(user_1);
+        records.add(user_2);
+
+        given(userRepository.findAll()).willReturn(records);
+        List<UserDto> users = userService.getAll();
+        Mockito.verify(userRepository, times(1)).findAll();
+        assertNotNull(users);
+        assertEquals(2, users.size());
+
+
+    }
+
+    @Test
+    void delete_WithNonExistentId_ThrowsException() {
+        assertThrows(NoSuchUserExistsException.class, () -> {
+            int id = 1;
+            User user = new User();
+            user.setId(id);
+            given(userRepository.findById(id)).willReturn(Optional.empty());
+            userService.delete(id);
+
+            verify(userRepository, never()).deleteById(id);
+        });
+
+    }
+
+    @Test
+    void delete_WithExistentId_DoNothing() {
+        int id = 1;
+        User user = new User();
+        user.setId(id);
+        given(userRepository.findById(id)).willReturn(Optional.of(user));
+        userService.delete(id);
+        verify(userRepository, times(1)).deleteById(id);
+
+
+    }
+
+    @Test
+    void getByUsername_GivenUsernameExists_ReturnsUser() {
+        String username = "taha.furkan";
+        User user = new User();
+        user.setUsername(username);
+        given(userRepository.findByUsername(username)).willReturn(user);
+        User user1 = userService.getByUsername(username);
+        assertEquals(username, user1.getUsername());
+    }
+
+    @Test
+    void getByUsername_GivenUsernameNonExists_DoNothing() {
+        String username = "taha.furkan";
+        User user = new User();
+        user.setUsername(username);
+        given(userRepository.findByUsername(username)).willReturn(null);
+        User user_1 = userService.getByUsername(username);
+        assertEquals(null, user_1);
+
+    }
```

Comment:
`user1 ` ve `user_1 ` seklindeki degisken isimlendirmelerini kullanma, ozellikle `user_1` seklinde isimlendirme Java'da olacak bir sey degil, daha once ne demistik, Java'da isimler _camelCase_ olmali, `_` kullanimi yok Java'da.

`getByUsername_GivenUsernameExists_ReturnsUser ` testini asagidaki gibi yazabilirsin:
```java
    @Test
    void getByUsername_GivenUsernameExists_ReturnsUser() {
        String username = "taha.furkan";
        User mockedUser = new User();
        mockedUser.setUsername(username);
        given(userRepository.findByUsername(username)).willReturn(mockedUser);
        User returnedUser = userService.getByUsername(username);
        assertEquals(username, returnedUser.getUsername());
    }
```
`getByUsername_GivenUsernameNonExists_DoNothing` testinde ise zaten bir `user` tanimlamana gerek yok, cunku kullanmiyorsun testin devaminda!
Daha once `assertNotNull`'i kullanmistik, burada da `assertNull`'i kullanabilirsin.
Bunu da asagidaki gibi yazabilirsin:
```java
    @Test
    void getByUsername_GivenUsernameNonExists_DoNothing() {
        String username = "taha.furkan";
        given(userRepository.findByUsername(username)).willReturn(null);
        User user = userService.getByUsername(username);
        assertNull(user);
    }
```

Reply:
**Not:** Bu arada Java'da degisken(variable) ve method isimlendirmeleri _camelCase_ olacak diyoruz ama test metotlarinda buna uymuyoruz dikkat edersen. `getByUsername_GivenUsernameExists_ReturnsUser` gibi.
Bu sadece test metotlarinin isimlendirmesine ozgu bir sey, test caseleri anlasilir olsun diye, baska bir yerde kullanilmaz bu sekilde.